lib/angular/directive.js

```diff
@@ -32,10 +39,35 @@ angular.module('wfm.file.directives', [])
       scope.$watch('uid', function(uid) {
         initPromise.then(function(cloudUrl) {
           element[0].src = cloudUrl + config.apiPath + '/get/' + uid;
-          console.log(element[0].src);
         });
       });
     }
   };
 })
+  .directive('fileDetail', function($templateCache) {
+    /*
+     detail view of a file, passed in is file object and options.
+     Options help to chose which details we want to render:
+     e.g. <file-detail file="ctrl.file" options="id, name, uid, owner, preview">
+     would render all details.
+     Options should be comma separated ', '
+     */
+    return {
+      restrict: 'E',
+      template: $templateCache.get('wfm-template/file-detail.tpl.html'),
+      scope: {
+        file: '=',
+        displayOptions: '='
+      },
+      controller: function($scope) {
+        var self = this;
+        self.file = $scope.file;
+        self.options = {};
+        _.forEach($scope.displayOptions, function(option) {
```

Comment:
Not sure why you need to define another "options" variable here.

You could have ```$scope.displayOptions = $scope.displayOptions || {}```

and then use that in your template.